javascripts/discourse/services/docs-sidebar.js

```diff
@@ -0,0 +1,144 @@
+  @service store;
+
+  #contentCache = new Map();
+
+  @tracked _activeTopicId;
+  @tracked _currentSectionsConfig = null;
+  @tracked _loading = false;
+
+  constructor() {
+    super(...arguments);
+
+    this.appEvents.on("page:changed", this, this.#maybeForceDocsSidebar);
+  }
+
+  get activeCategory() {
+    return (
+      this.router.currentRoute?.attributes?.category ||
+      this.router.currentRoute?.parent?.attributes?.category
+    );
+  }
+
+  get allSectionsExpanded() {
+    return this.sectionsConfig?.every((sectionConfig) => {
+      return !this.sidebarState.collapsedSections.has(
+        `sidebar-section-${sectionConfig.name}-collapsed`
+      );
+    });
+  }
+
+  get isEnabled() {
+    return !!this._activeTopicId;
+  }
+
+  get isVisible() {
+    return this.sidebarState.isCurrentPanel(SIDEBAR_DOCS_PANEL);
+  }
+
+  get loading() {
+    return this._loading;
+  }
+
+  get sectionsConfig() {
+    return this._currentSectionsConfig || [];
+  }
+
+  hideDocsSidebar() {
+    this.sidebarState.setPanel(MAIN_PANEL);
+  }
+
+  showDocsSidebar() {
+    this.sidebarState.setPanel(SIDEBAR_DOCS_PANEL);
+    this.sidebarState.setSeparatedMode();
+    this.sidebarState.hideSwitchPanelButtons();
+  }
+
+  toggleSidebarPanel() {
+    if (this.isVisible) {
+      this.hideDocsSidebar();
+    } else {
+      this.showDocsSidebar();
+    }
+  }
+
+  #findSettingsForActiveCategory() {
+    let category = this.activeCategory;
+
+    while (category != null) {
+      const matchingSetting = settings.categories_toc_topics.find(
+        // eslint-disable-next-line no-loop-func
+        (setting) => setting.category[0] === category.id
+      );
+
+      if (matchingSetting) {
+        return matchingSetting;
+      }
+
+      category = category.parentCategory;
+    }
+  }
+
+  #maybeForceDocsSidebar() {
+    const newActiveTopicId = this.#findSettingsForActiveCategory()?.topic_id;
+
+    if (this._activeTopicId !== newActiveTopicId) {
+      this.#setSidebarContent(newActiveTopicId);
+    }
+  }
+
+  async #setSidebarContent(topic_id) {
+    this._activeTopicId = topic_id;
+
+    if (!this._activeTopicId) {
+      this.hideDocsSidebar();
+      return;
+    }
+
+    this._currentSectionsConfig = this.#contentCache.get(this._activeTopicId);
+
+    if (this._currentSectionsConfig) {
+      this.showDocsSidebar();
+      return;
+    }
+
+    await this.#fetchTopicContent(topic_id);
+  }
+
+  async #fetchTopicContent(topic_id) {
+    this._loading = true;
+    this.showDocsSidebar();
+
+    try {
+      // leverages the post stream API to fetch only the first post
+      const data = await ajax(`/t/${topic_id}/posts.json`, {
+        post_number: 2,
+        include_suggested: false,
+        asc: false,
+      });
+
+      const cookedHtml = data?.post_stream?.posts?.[0]?.cooked;
+      if (!cookedHtml) {
+        // display regular sidebar
+        return;
+      }
+
+      const sections = parseSidebarStructure(cookedHtml);
```

Comment:
What happens if the parser fails to extract the HTML correctly?

Reply:
I'm not sure whether this is the behavior we will want, but for now, I added a commit to make the component fallback to the default sidebar if there is an error while fetching/parsing the content.

It will also fall back to the sidebar if the topic parsed does not contain at least a section with a link matching the expected format

Reply:
Yep makes sense to fall back to the default for now :+1: